src/sharepoint/rest/search.ts

```diff
@@ -0,0 +1,429 @@
+
+    /** 
+     * The GUID for the user who submitted the search query.
+     */
+    PersonalizationData?: string;
+
+    /** 
+     * The URL for the search results page.
+     */
+    ResultsURL?: string;
+
+    /** 
+     * Custom tags that identify the query. You can specify multiple query tags
+     */
+    QueryTag?: string[];
+
+    // TODO: Properties
+
+    // TODO: ReorderingRules
+
+    /**
+     *  A Boolean value that specifies whether to return personal favorites with the search results.
+     */
+    ProcessPersonalFavorites?: Boolean;
+
+    /**
+     * The location of the queryparametertemplate.xml file. This file is used to enable anonymous users to make Search REST queries.
+     */
+    QueryTemplatePropertiesUrl?: string;
+
+    /**
+     * Special rules for reordering search results. 
+     * These rules can specify that documents matching certain conditions are ranked higher or lower in the results. 
+     * This property applies only when search results are sorted based on rank. 
+     */
+    ReorderingRules?: ReorderingRule[];
+
+    /**
+     * The number of properties to show hit highlighting for in the search results.
+     */
+    HitHighlightedMultivaluePropertyLimit?: Number;
+
+    /**
+     * A Boolean value that specifies whether the hit highlighted properties can be ordered.
+     */
+    EnableOrderingHitHighlightedProperty?: Boolean;
+
+    /**
+     * The managed properties that are used to determine how to collapse individual search results. 
+     * Results are collapsed into one or a specified number of results if they match any of the individual collapse specifications. 
+     * In a collapse specification, results are collapsed if their properties match all individual properties in the collapse specification.
+     */
+    CollapseSpecification?: String;
+
+    /**
+     * The locale identifier (LCID) of the user interface
+     */
+    UIlanguage?: Number;
+
+    /**
+     * The preferred number of characters to display in the hit-highlighted summary generated for a search result.
+     */
+    DesiredSnippetLength?: Number;
+
+    /**
+     * The maximum number of characters to display in the hit-highlighted summary generated for a search result.
+     */
+    MaxSnippetLength?: Number;
+
+    /**
+     * The number of characters to display in the result summary for a search result.
+     */
+    SummaryLength?: Number;
+
+}
+
+/**
+ * Describes the search API
+ * 
+ */
+export class Search extends QueryableInstance {
+
+    /**
+     * Creates a new instance of the Search class
+     * 
+     * @param baseUrl The url for the search context
+     * @param query The SearchQuery object to execute
+     */
+    constructor(baseUrl: string | Queryable, query: SearchQuery) {
+        super(baseUrl, "postquery");
+
+        this.searchQuery = query;
+    }
+
+    private searchQuery: SearchQuery = null;
+
+    /**
+     * .......
+     * @returns Promise
+     */
+    public get(): Promise<any> {
```

Comment:
This is me being me, but can you rename this to like "execute" or something similar? Then just call this method in the search() above instead of returning the search, return this result (a promise) directly. Small change, but I think makes it even cleaner.